state-sync/state-sync-v2/state-sync-driver/src/tests/bootstrapper.rs

```diff
@@ -429,6 +430,369 @@ async fn test_fetch_epoch_ending_ledger_infos() {
+        .returning(|_| Ok(true));
+
+    // Create the bootstrapper
+    let mut bootstrapper = create_bootstrapper_with_storage(
+        driver_configuration,
+        mock_streaming_client,
+        metadata_storage,
+        synced_version,
+        true,
+    );
+
+    // Insert an epoch ending ledger info into the verified states of the bootstrapper
+    manipulate_verified_epoch_states(&mut bootstrapper, true, true, Some(highest_version));
+
+    // Create a global data summary
+    let mut global_data_summary = create_global_summary(1);
+    global_data_summary.advertised_data.synced_ledger_infos = vec![highest_ledger_info.clone()];
+
+    // Drive progress and verify that the bootstrapper panics (due to invalid state)
+    drive_progress(&mut bootstrapper, &global_data_summary, false)
+        .await
+        .unwrap();
+}
+
+#[tokio::test]
+async fn test_snapshot_sync_lag() {
+    // Create test data
+    let num_versions_behind = 1000;
+    let highest_version = 1000000;
+    let synced_version = highest_version - num_versions_behind;
+    let highest_ledger_info = create_random_epoch_ending_ledger_info(highest_version, 1);
+
+    // Create a driver configuration with a genesis waypoint and state syncing
+    let mut driver_configuration = create_full_node_driver_configuration();
+    driver_configuration.config.bootstrapping_mode = BootstrappingMode::DownloadLatestStates;
+    driver_configuration
+        .config
+        .num_versions_to_skip_snapshot_sync = num_versions_behind + 1;
+
+    // Create the mock streaming client
+    let mock_streaming_client = create_mock_streaming_client();
+
+    // Create the mock metadata storage
+    let mut metadata_storage = MockMetadataStorage::new();
+    metadata_storage
+        .expect_previous_snapshot_sync_target()
+        .returning(|| Ok(None));
+
+    // Create the bootstrapper
+    let mut bootstrapper = create_bootstrapper_with_storage(
+        driver_configuration,
+        mock_streaming_client,
+        metadata_storage,
+        synced_version,
+        true,
+    );
+
+    // Insert an epoch ending ledger info into the verified states of the bootstrapper
+    manipulate_verified_epoch_states(&mut bootstrapper, true, true, Some(highest_version));
+
+    // Create a global data summary
+    let mut global_data_summary = create_global_summary(1);
+    global_data_summary.advertised_data.synced_ledger_infos = vec![highest_ledger_info.clone()];
+
+    // Drive progress to mark bootstrapping complete (we're within the snapshot sync lag)
+    drive_progress(&mut bootstrapper, &global_data_summary, false)
+        .await
+        .unwrap();
+
+    // Verify the bootstrapper has completed
+    assert!(bootstrapper.is_bootstrapped());
+}
+
+#[tokio::test]
+#[should_panic(
+    expected = "Snapshot syncing is currently unsupported for nodes with existing state!"
+)]
+async fn test_snapshot_sync_lag_panic() {
+    // Create test data
+    let num_versions_behind = 10000;
```

Comment:
same as above

Reply:
This is not the same value as the config. This is the number of versions we're actually behind. The config is set to be lower than this (see below).

Reply:
I think is cleaner as it actually checks that the config values are being used in the code 😄